static/monaco-config.ts

```diff
@@ -39,19 +38,39 @@ var config = {
     emptySelectionClipboard: true,
 };
 
-function extendConfig(options, settings) {
-    var settingsObject = {};
+type EditorKinds =
+    | monaco.editor.IStandaloneDiffEditor
+    | monaco.editor.IStandaloneCodeEditor;
+
+/** Pick consturction options based on editor kind */
+type EditorConstructionType<E extends EditorKinds> =
+    E extends monaco.editor.IStandaloneDiffEditor
+        ? monaco.editor.IDiffEditorConstructionOptions
+        : monaco.editor.IStandaloneEditorConstructionOptions;
+
+/**
+ * Extend the default monaco editor construction options.
+ *
+ * Type parameter E indicates which editor kind you're constructing a config
+ * for. Valid options are EditorKinds, aka monaco.editor.IStandaloneDiffEditor
+ * or monaco.editor.IStandaloneCodeEditor
+ *
+ * TODO(supergrecko): underscore.extend yields any, can we improve the type
+ *  check on this?
+ */
+export function extendConfig<
+    E extends EditorKinds = monaco.editor.IStandaloneCodeEditor,
+    T = EditorConstructionType<E>>(
+    overrides: T,
+    settings?: any, /* SiteSettings #2917 */
```

Comment:
This should be typed as the compiler explorer site settings which does not have a type at the moment

Reply:
Depends on #2917 which adds a typed interface to the site settings